### PR TITLE
Fix Schedule Helper UI

### PR DESCRIPTION
**Pull request by zsarnett**
## Proposed change

- Fix for 12 AM / 24:00 / 0:00 ending times
- Allows for dragging to midnight
- Caps selections that span over days to midnight
- fixes locale on event and side bar
- wraps event time

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/helpers/forms/ha-schedule-form.ts [hunk 1]
```diff
@@ -234,65 +240,64 @@ class HaScheduleForm extends LitElement {
 
   private _handleSelect(info: { start: Date; end: Date }) {
     const { start, end } = info;
-
-    if (start.getDay() !== end.getDay()) {
-      this.calendar!.unselect();
-      return;
-    }
+    const dayDiff = Math.abs(start.getDay() - end.getDay());
```
[the remark this answers is not in the document]
Comment (bramkragten) [reply src/panels/config/helpers/forms/ha-schedule-form.ts:243]: Or in this case maybe it should be `isSameDay`?

File: src/panels/config/helpers/forms/ha-schedule-form.ts [hunk 2]
```diff
@@ -274,21 +276,24 @@ class HaScheduleForm extends LitElement {
...
-    if (dayDiff >= 1) {
+    if (isSameDay(start, end)) {
       info.revert();
     }
   }
 
   private _handleEventDrop(info: any) {
     const { id, start, end } = info.event;
-    const dayDiff = Math.abs(start.getDay() - end.getDay());
+    isSameDay(start, end);
```
Comment (bramkragten) [inline src/panels/config/helpers/forms/ha-schedule-form.ts:296]: right? 😄

Comment (zsarnett) [reply src/panels/config/helpers/forms/ha-schedule-form.ts:296]:
```suggestion
```

File: src/common/datetime/format_time.ts [hunk 3]
```diff
@@ -71,7 +71,8 @@ export const formatTime24h = (dateObj: Date) =>
 
 const formatTime24hMem = memoizeOne(
   () =>
-    new Intl.DateTimeFormat(undefined, {
+    // en-GB to fix Chrome 24:59 to 0:59 https://stackoverflow.com/a/60898146
+    new Intl.DateTimeFormat("en-GB", {
```
[the remark this answers is not in the document]
Comment (balloob) [reply src/common/datetime/format_time.ts:75]: Probably only use `en-GB` if the localse is `en` ?

[the remark this answers is not in the document]
Comment (zsarnett) [reply src/common/datetime/format_time.ts:75]: This is just to format it for the backend

[the remark this answers is not in the document]
Comment (zsarnett) [reply src/common/datetime/format_time.ts:75]: The backend uses `en`

[the remark this answers is not in the document]
Comment (zsarnett) [reply src/common/datetime/format_time.ts:75]: Its also just a 24 hour format. its the same everywhere. The date time numbers will be diff but not the format

[the remark this answers is not in the document]
Comment (balloob) [reply src/common/datetime/format_time.ts:75]: oh right 24hour is in the function name 🤦

**Review by balloob — APPROVED** [review]

**Pull request merged**